### PR TITLE
SDK-170 Java BDK: Create test coverage for com.symphony.bdk.bot.sdk.monitoring.SymphonyHealthIndicator

### PR DESCRIPTION
**Pull request by Haiddi-Avencod**
### Ticket
[SDK-170](https://perzoinc.atlassian.net/browse/SDK-170?atlOrigin=eyJpIjoiOTEyM2YxNzI1MmMxNDhmMmIyMDBlNDVjMzE2ZWQ0NGEiLCJwIjoiaiJ9)

### Coverage
coverage_170

### Comment
Folder `Mockito-extensions` with file `org.mockito.plugins.MockMaker` added in `resources` folder to mock final classes

### CODE REVIEW

File: symphony-bdk-legacy/symphony-bdk-bot-sdk-java/src/test/java/com/symphony/bdk/bot/sdk/monitoring/SymphonyHealthIndicatorTest.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  public void initSymphonyHealthIndicator(){
+    this.initRestClient();
+    this.initSymBotClient();
+    this.healthcheckClient = new HealthcheckClientImpl(this.symBotClient, this.restClient);
+    this.symphonyHealthIndicator = new SymphonyHealthIndicator(this.healthcheckClient);
+  }
+
+  @Test
+  public void testHealth(){
+    final Health health = this.symphonyHealthIndicator.health();
```
Comment (symphony-elias) [inline symphony-bdk-legacy/symphony-bdk-bot-sdk-java/src/test/java/com/symphony/bdk/bot/sdk/monitoring/SymphonyHealthIndicatorTest.java:43]: Could you as well test the value of `health.getStatus()`? There are two possible values (UP and DOWN), so please create another test case for the second value.

Comment (Haiddi-Avencod) [reply symphony-bdk-legacy/symphony-bdk-bot-sdk-java/src/test/java/com/symphony/bdk/bot/sdk/monitoring/SymphonyHealthIndicatorTest.java:43]: When the method `checkOverallStatus()` returns `false` we can verify the status is `Down`, but I can't find the necessary condition for status to be `UP`.

Comment (symphony-elias) [reply symphony-bdk-legacy/symphony-bdk-bot-sdk-java/src/test/java/com/symphony/bdk/bot/sdk/monitoring/SymphonyHealthIndicatorTest.java:43]: Simplest way would be to mock to `HealthcheckClient` and pass the mock instance to the constructor of  `SymphonyHealthIndicator`. This way, you won't have to instantiate `SymBotClient` and `RestClient`.

Comment (Haiddi-Avencod) [reply symphony-bdk-legacy/symphony-bdk-bot-sdk-java/src/test/java/com/symphony/bdk/bot/sdk/monitoring/SymphonyHealthIndicatorTest.java:43]: I don't understand the link with my problem ...

Comment (Haiddi-Avencod) [reply symphony-bdk-legacy/symphony-bdk-bot-sdk-java/src/test/java/com/symphony/bdk/bot/sdk/monitoring/SymphonyHealthIndicatorTest.java:43]: That's good, sorry, I've just understood ;)

**Review by symphony-elias — APPROVED** [review]
Looks good ! 👍

**Pull request merged**